src/Core/Geographic/Coordinates.js

```diff
@@ -224,8 +224,32 @@ export function convertValueToUnit(unitIn, unitOut, value) {
 function Coordinates(crs, ...coordinates) {
     this._values = new Float64Array(3);
     this.set(crs, ...coordinates);
+
+    Object.defineProperty(this, 'surfaceNormal',
+        {
+            configurable: true,
+            get: () => {
+                const result = this._normal || this.computeSurfaceNormal();
+                return result;
+            },
+        });
 }
 
+Coordinates.prototype.computeSurfaceNormal = function computeSurfaceNormal() {
+    // compute a normal only for cartesian coordinate.
+    if (this._internalStorageUnit != UNIT.METER) {
```

Comment:
Please don't throw an error.
You just need to handle the `4326` case (using `ellipsoid.cartographicToCartesian`) and you're all set.